packages/date/src/index.js

```diff
@@ -226,15 +226,31 @@ export function __experimentalGetSettings() {
 }
 
 function setupWPTimezone() {
-	// Create WP timezone based off dateSettings.
-	momentLib.tz.add(
-		momentLib.tz.pack( {
-			name: WP_ZONE,
-			abbrs: [ WP_ZONE ],
-			untils: [ null ],
-			offsets: [ -settings.timezone.offset * 60 || 0 ],
-		} )
-	);
+	// Get the current timezone settings from the WP timezone string.
+	const currentTimeZone = momentLib.tz.zone( settings.timezone.string );
+
+	// Check to see if we have a valid TZ data, if so, use it fro the custom WP_ZONE timezone, otherwise just use the offset.
+	if ( currentTimeZone ) {
+		// Create WP timezone based off dateSettings.
+		momentLib.tz.add(
+			momentLib.tz.pack( {
+				name: WP_ZONE,
+				abbrs: currentTimeZone.abbrs,
+				untils: currentTimeZone.untils,
+				offsets: currentTimeZone.offsets,
+			} )
+		);
+	} else {
+		// Create WP timezone based off dateSettings.
+		momentLib.tz.add(
+			momentLib.tz.pack( {
+				name: WP_ZONE,
+				abbrs: [ WP_ZONE ],
+				untils: [ null ],
+				offsets: [ -settings.timezone.offset * 60 || 0 ],
+			} )
+		);
+	}
```

Comment:
Adding two minor suggestions:

1. Rename `currentTimeZone` to `currentTimezone` since everywhere else treats "timezone" as a single word instead of two.
2. Update comments to reflect discussion in this PR. This is more of a personal philosophy than a WP one, but I find comments that explain "why I wrote it this way" much more useful than ones that explain "what I wrote".

```suggestion
	const currentTimezone = momentLib.tz.zone( settings.timezone.string );

	// Prefer the timezone string over the offset like other places using both.
	if ( currentTimezone ) {
		// Use add() instead of link() because add() will override the
		// existing WP_ZONE if it is already set.
		momentLib.tz.add(
			momentLib.tz.pack( {
				name: WP_ZONE,
				abbrs: currentTimezone.abbrs,
				untils: currentTimezone.untils,
				offsets: currentTimezone.offsets,
			} )
		);
	} else {
		momentLib.tz.add(
			momentLib.tz.pack( {
				name: WP_ZONE,
				abbrs: [ WP_ZONE ],
				untils: [ null ],
				offsets: [ -settings.timezone.offset * 60 || 0 ],
			} )
		);
	}
```

Reply:
Done.